src/main/java/racing/core/domain/Cars.java

```diff
@@ -26,11 +22,12 @@ private void validateList(List<Car> cars) {
         }
     }
 
-    public Trial nextTrial(MoveStrategy movement) {
-        List<TrackInfo> tracks = cars.stream()
-                .map(car -> car.move(movement))
-                .collect(Collectors.toList());
-        return new Trial(tracks);
+    public Cars runTrial(MoveStrategy movement) {
+        List<Car> after = new ArrayList<>();
+        for (Car car : cars) {
+            after.add(car.move(movement));
+        }
```

Comment:
Stream을 활용하면 아래처럼 바꿀 수 있습니다 :)
```suggestion
        List<Car> afterCars = cars.stream()
                .map(car -> car.move(movement))
                .collect(Collectors.toList());
```